pages/thank-you.jsx

```diff
@@ -1,27 +1,40 @@
-// this will ignore that Bad Invocation error which is a bug in JSXHint
-/* jshint -W067 */
-
 import React from 'react';
 import MozillaFooter from '../components/footer-mozilla.jsx';
 import Signup from '../components/signup.jsx';
 import Social from '../components/social.jsx';
 import ThankYouHeader from '../components/thank-you-header.jsx';
-import { IntlMixin } from 'react-intl';
 import analytics from '../assets/js/analytics.js';
 import form from '../scripts/form.js';
+import qs from 'qs';
 
 var ThankYou = React.createClass({
-  mixins: [IntlMixin],
+  contextTypes: {
+    location: React.PropTypes.object,
+    router: React.PropTypes.object,
+    intl: React.PropTypes.object
+  },
+  propTypes: {
+    country: React.PropTypes.string.isRequired,
+    test: React.PropTypes.string,
+    location: React.PropTypes.object
+  },
+  getInitialState: function() {
+    let query = qs.parse(this.props.location.search.replace("?", ""));
```

Comment:
The problem is how react router handles routes server side and thus we don't have access to that yet unless we parse that and make it an object. The way we used to do it was client side only that's why